lib/importproject.cpp

```diff
@@ -1149,8 +1149,8 @@ bool ImportProject::importCppcheckGuiProject(std::istream &istr, Settings *setti
         else if (strcmp(node->Name(), CppcheckXml::DefinesElementName) == 0)
             temp.userDefines = join(readXmlStringList(node, "", CppcheckXml::DefineName, CppcheckXml::DefineNameAttrib), ";");
         else if (strcmp(node->Name(), CppcheckXml::UndefinesElementName) == 0) {
-            for (const std::string &u : readXmlStringList(node, "", CppcheckXml::UndefineName, nullptr))
-                temp.userUndefs.insert(u);
+            for (auto &u : readXmlStringList(node, "", CppcheckXml::UndefineName, nullptr))
```

Comment:
Is this safe? A nonconst reference to a temporary does not ensure that the object remains alive in the next statement. If it's not safe.. I wonder if cppcheck warns correctly..

Reply:
personally I prefer `std::string` is written instead of `auto`. It's not obvious that readXmlStringList returns a std::string. It sounds like it could return a std::vector for instance.

Reply:
> Is this safe? A nonconst reference to a temporary does not ensure that the object remains alive in the next statement. If it's not safe.. I wonder if cppcheck warns correctly..

Interesting. Something to look into... :|
I checked the web on how to do this and they suggested using `auto&` which did not work in another case since that resulted in `const T&` (as noted by clang-tidy). Another case where modern C++ is horrible without proper tooling.

> personally I prefer `std::string` is written instead of `auto`. It's not obvious that readXmlStringList returns a std::string. It sounds like it could return a std::vector for instance.

Agreed.

I will just revert this case. It always felt awkward and it's not in a hot path.